web/src/components/SignUpDialogs.tsx

```diff
@@ -0,0 +1,225 @@
+import Config from "@businessnjgovnavigator/content/fieldConfig/config.json";
+import { RegistrationStatus } from "@businessnjgovnavigator/shared";
+import { Box, Dialog, DialogContent, DialogTitle, IconButton } from "@mui/material";
+import { useRouter } from "next/router";
+import React, { ReactElement, ReactNode, useContext, useEffect } from "react";
+
+export const SelfRegToast = (): ReactElement => {
+  const { isAuthenticated, registrationAlertStatus, setRegistrationAlertStatus } =
+    useContext(AuthAlertContext);
+
+  useEffect(() => {
+    if (registrationAlertStatus == "IN_PROGRESS" && isAuthenticated == IsAuthenticated.TRUE) {
+      setRegistrationAlertStatus("SUCCESS");
+    }
+  }, [isAuthenticated, registrationAlertStatus, setRegistrationAlertStatus]);
+
+  const alertMap: Record<Exclude<RegistrationStatus, "IN_PROGRESS">, AlertVariant> = {
+    SUCCESS: "success",
+    DUPLICATE_ERROR: "error",
+    RESPONSE_ERROR: "error",
+  };
+
+  const contentMap: Record<Exclude<RegistrationStatus, "IN_PROGRESS">, string> = {
+    SUCCESS: Config.navigationDefaults.guestSuccessBody,
+    DUPLICATE_ERROR: Config.selfRegistration.errorTextDuplicateSignup,
+    RESPONSE_ERROR: Config.selfRegistration.errorTextGeneric,
+  };
+
+  if (registrationAlertStatus && registrationAlertStatus != "IN_PROGRESS") {
+    return (
+      <ToastAlert
+        isOpen={!!registrationAlertStatus}
+        close={() => setRegistrationAlertStatus(undefined)}
+        variant={alertMap[registrationAlertStatus]}
+        noIcon={true}
+        autoHideDuration={null}
+        snackBarProps={{ sx: { paddingX: 0 } }}
+      >
+        <div className="fin fac padding-y-2" data-testid={"reg-toast"}>
+          {alertMap[registrationAlertStatus] == "success" ? (
+            <img
+              src={`/img/congratulations-green.svg`}
+              alt="congratulations"
+              data-testid={"congratulations-logo"}
+              style={{ marginRight: "20px", width: "64px", height: "64px" }}
+            />
+          ) : (
+            <></>
+          )}
+          <div>
+            {alertMap[registrationAlertStatus] == "success" ? (
+              <Content className="padding-right-2">{Config.navigationDefaults.guestSuccessTitle}</Content>
+            ) : (
+              <></>
+            )}
+            <IconButton
+              aria-label="close"
+              onClick={() => setRegistrationAlertStatus(undefined)}
+              sx={{
+                position: "absolute",
+                right: 0,
+                top: 0,
+                color: (theme) => theme.palette.grey[500],
+              }}
+            >
+              <Icon className="usa-icon--size-4">close</Icon>
+            </IconButton>
+            <Content>{contentMap[registrationAlertStatus]}</Content>
+          </div>
+        </div>
+      </ToastAlert>
+    );
+  }
+  return <></>;
+};
+export const UseAuthToast = (): ReactElement => {
```

Comment:
can we separate these into different files? this makes them harder to search for and find later

Reply:
same with the tests

Reply:
also, naming a component with the `use` prefix feels weird, that should be reserved for hooks